src/documentation/search/SearchResultList.tsx

```diff
@@ -37,38 +44,45 @@ const SearchResultList = ({
 
 interface SearchResultItemProps {
   value: Result;
-  onClose: () => void;
+  viewedResults: string[];
+  onViewResult: (id: string, navigation: RouterState) => void;
 }
 
 const SearchResultItem = ({
-  value: { extract, navigation, containerTitle, title },
-  onClose,
+  value: { extract, navigation, containerTitle, title, id },
+  viewedResults,
+  onViewResult,
 }: SearchResultItemProps) => {
-  const [, setState] = useRouterState();
+  const url = toUrl(getQueryString(navigation), navigation);
```

Comment:
Let's move the getQueryString call inside toUrl so the caller just gets a URL from the router state and doesn't have to make two calls.

Reply:
[`a905f61`](https://github.com/microbit-foundation/python-editor-next/pull/477/commits/a905f6178a9918ecaed74cf31ef8b850b8005960)

Reply:
Change looks great, will just wait for it to build then merge.